lib/SearchProvider.js

```diff
@@ -134,7 +132,7 @@ NewTabSearchProvider.prototype = {
   /**
    *  Gets the state - a combination of the current engine and all the visible engines.
    */
-  asyncGetCurrentState: Task.async(function*() {
+  get currentState() {
```

Comment:
This is the change in question. If we do want to keep this async, we can have eslint ignore this and undo the other changes related to `asyncGetCurrentState` -> `currentState`.